dttools/src/bucketing.c

```diff
@@ -142,32 +140,47 @@ bucketing_bucket_t* bucketing_bucket_create(double val, double prob)
 
 void bucketing_bucket_delete(bucketing_bucket_t* b)
 {
-    if (b)
-        free(b);
-    else
-        warn(D_BUCKETING, "ignoring command to delete a null pointer to a bucket\n");
+    free(b);
 }
 
 bucketing_state_t* bucketing_state_create(double default_value, int num_sampling_points,
     double increase_rate, int max_num_buckets, bucketing_mode_t mode, int update_epoch)
 {
     if (default_value < 0)
+    {
         warn(D_BUCKETING, "default value cannot be less than 0\n");
+        default_value = 1;
+    }
 
     if (num_sampling_points < 1)
+    {
         warn(D_BUCKETING, "number of sampling points cannot be less than 1\n");
+        num_sampling_points = 1;
+    }
 
     if (increase_rate <= 1)
+    {
         warn(D_BUCKETING, "increase rate must be greater than 1 to be meaningful\n");
+        increase_rate = 2;
+    }
 
     if (max_num_buckets < 1 && mode == BUCKETING_MODE_EXHAUSTIVE)
+    {
         warn(D_BUCKETING, "The maximum number of buckets for exhaustive bucketing must be at least 1\n");
+        max_num_buckets = 1;
+    }
     
     if (mode != BUCKETING_MODE_GREEDY && mode != BUCKETING_MODE_EXHAUSTIVE)
+    {
         fatal("Invalid bucketing mode\n");
+        mode = BUCKETING_MODE_GREEDY;
```

Comment:
fatal exits the program. Do you mean warn? If you mean fatal, then line 176 can be removed.